websocket.go

```diff
@@ -51,3 +55,28 @@ var wsServe = func(cfg *wsConfig, handler WsHandler, errHandler ErrHandler) (don
 	}()
 	return
 }
+
+func keepAlive(c *websocket.Conn, timeout time.Duration) {
+	ticker := time.NewTicker(timeout)
+
+	lastResponse := time.Now()
+	c.SetPongHandler(func(msg string) error {
+		lastResponse = time.Now()
+		return nil
+	})
+
+	go func() {
+		defer ticker.Stop()
+		for {
+			err := c.WriteMessage(websocket.PingMessage, []byte("keepalive"))
```

Comment:
use c.WriteControl to send a ping message without data: https://godoc.org/github.com/gorilla/websocket#Conn.WriteControl